controller: controller_execute: delete client output and chomp output from publish hook before sending to subscribe hooks

Miciah opened the pull request:
#### controller: `execute_connections`: delete and chomp client output

Filter client output out from the publish hooks output and chomp it before providing it as input to the subscribe hooks.

This commit fixes bug 1261540.
#### controller: `execute_connections`: reformat and comment

Reformat `execute_connections` to return right away if the application is not scalable rather than wrapping the entire function in an `if` block.

Reformat the block that calls the publish hooks to go to the next hook right away if the current hook did not run successfully rather than wrapping the entire block in an `if` block.

Add some comments.
#### Log errors and skip bad lines when writing environment variables

`write_environment_variables`: Rescue and log exceptions raised when writing out the file.

`set_connection_hook_env_vars`: Skip malformed lines that do not include the '=' key-value separator.

Miciah committed:
controller: execute_connections: nix client output

Filter client output out from the publish hooks output and chomp it
before providing it as input to the subscribe hooks.

This commit fixes bug 1261540.

dobbymoodge commented on controller/app/models/application.rb:
```diff
@@ -1595,6 +1596,8 @@ def execute_connections
           else
             pub_out[tag] = [] if pub_out[tag].nil?
+            re = /^CLIENT_(MESSAGE|RESULT|DEBUG|ERROR|INTERNAL_ERROR)/
+            output = output.lines.reject {|line| line =~ re}.join
```
Shouldn't `output.chomp!('')` be before this line?

dobbymoodge commented on controller/app/models/application.rb:
```diff
@@ -1588,6 +1588,7 @@ def execute_connections
       pub_out = {}
       RemoteJob.execute_parallel_jobs(handle)
       RemoteJob.get_parallel_run_results(handle) do |tag, gear_id, output, status|
+        output.chomp!
```
Wrong place for this? see below

Miciah committed:
controller: execute_connections: reformat, comment

Reformat execute_connections to return right away if the application is
not scalable rather than wrapping the entire function in an if block.

Reformat the block that calls the publish hooks to go to the next hook
right away if the current hook did not run successfully rather than
wrapping the entire block in an if block.

Add some comments.

Miciah committed:
Log errs and skip bad lines when writing env vars

write_environment_variables: Rescue and log exceptions raised when writing
out the file.

set_connection_hook_env_vars: Skip malformed lines that do not include the
'=' key-value separator.

dobbymoodge commented on controller/app/models/application.rb:
```diff
@@ -1566,67 +1566,93 @@ def get_unsubscribe_info(comp_inst)
+      sub_inst = self.component_instances.find(conn.to_comp_inst_id)
+      tag = ""
+
+      unless pub_out[conn._id.to_s].nil?
```
Crashes right here on my devenv

dobbymoodge replied:
Never reaches this point on my devenv

dobbymoodge commented on controller/app/models/application.rb:
```diff
@@ -1566,67 +1566,93 @@ def get_unsubscribe_info(comp_inst)
+      pub_inst = self.component_instances.find(conn.from_comp_inst_id)
+      sub_inst = self.component_instances.find(conn.to_comp_inst_id)
+      tag = ""
+
```
reaches THIS point on my devenv

Miciah commented:
Thanks!
